Build Error on ArchLinux

I get the following build error when trying to build it on ArchLinux.

```
g++ -c -pipe -O2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=gnu++11 -Wall -W -fno-exceptions -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050900 -DQT_NO_CAST_TO_ASCII -DQT_NO_PROCESS_COMBINED_ARGUMENT_START -DGIT_REVISION=\"alpha13-26-g1baa4e5\" -DGIT_DATE=\"2019-10-23\" -DQT_NO_EXCEPTIONS -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_MULTIMEDIA_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GAMEPAD_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_TESTLIB_LIB -DQT_SQL_LIB -DQT_CORE_LIB -DQT_TESTCASE_BUILDDIR='"/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/build/tests/backend/providers/favorites"' -I/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/tests/backend/providers/favorites -I. -I/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/src/backend -I/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/thirdparty -isystem /usr/include/qt -isystem /usr/include/qt/QtQuick -isystem /usr/include/qt/QtMultimedia -isystem /usr/include/qt/QtSvg -isystem /usr/include/qt/QtWidgets -isystem /usr/include/qt/QtGamepad -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtQml -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtTest -isystem /usr/include/qt/QtSql -isystem /usr/include/qt/QtCore -I. -I/usr/lib/qt/mkspecs/linux-g++ -o test_FavoriteDB.o /home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/tests/backend/providers/favorites/test_FavoriteDB.cpp
make[4]: Leaving directory '/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/build/tests/backend/providers/pegasus'
/usr/bin/ld: test_Collection.o:(.data.rel.ro._ZTI19QQmlObjectListModelIN5model4GameEE[_ZTI19QQmlObjectListModelIN5model4GameEE]+0x10): undefined reference to `typeinfo for QQmlObjectListModelBase'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:70: test_Collection] Error 1
make[4]: Leaving directory '/home/severin/dev/projects/PKGBUILDS/pegasus-frontend-git/src/pegasus-frontend/build/tests/backend/model/collection'
make[3]: *** [Makefile:54: sub-collection-make_first] Error 2
make[3]: *** Waiting for unfinished jobs....
```

On my Manjaro i also have an error at build : 

/usr/bin/ld: test_Collection.o:(.data.rel.ro._ZTI19QQmlObjectListModelIN5model4GameEE[_ZTI19QQmlObjectListModelIN5model4GameEE]+0x10): undefined reference to `typeinfo for QQmlObjectListModelBase'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:70: test_Collection] Error 1
make: *** [Makefile:73: sub-tests-make_first] Error 2
==> ERROR: A failure occurred in build().
    Aborting...

--- a/.qmake.conf
+++ b/.qmake.conf
@@ -2,22 +2,22 @@ TOP_SRCDIR=$$PWD
 TOP_BUILDDIR=$$shadowed($$PWD)
 
 GIT_REVISION = $$system(git \
-    --git-dir $${TOP_SRCDIR}/.git \
-    --work-tree $${TOP_SRCDIR} \
+    --git-dir \"$${TOP_SRCDIR}/.git\" \
+    --work-tree \"$${TOP_SRCDIR}\" \
     describe \
     --always --dirty)
 
 GIT_DATE = $$system(git \
-    --git-dir $${TOP_SRCDIR}/.git \
-    --work-tree $${TOP_SRCDIR} \
+    --git-dir \"$${TOP_SRCDIR}/.git\" \
+    --work-tree \"$${TOP_SRCDIR}\" \
     log \
     -1 \
     --format=%cd \
     --date=short)
 
 GIT_COMMIT_CNT = $$system(git \
-    --git-dir $${TOP_SRCDIR}/.git \
-    --work-tree $${TOP_SRCDIR} \
+    --git-dir \"$${TOP_SRCDIR}/.git\" \
+    --work-tree \"$${TOP_SRCDIR}\" \
     rev-list \
     --count HEAD)
 

--- a/src/app/app.pro
+++ b/src/app/app.pro
@@ -1,14 +1,11 @@
-TEMPLATE = app
 TARGET = pegasus-fe
-CONFIG += exceptions_off rtti_off
+CONFIG += c++11 warn_on exceptions_off rtti_off
 
 SOURCES += main.cpp
-RESOURCES += \
-    $${TOP_SRCDIR}/assets/assets.qrc
-OTHER_FILES += \
-    qmlplugins.qml
-
 DEFINES *= $${COMMON_DEFINES}
+
+RESOURCES += $${TOP_SRCDIR}/assets/assets.qrc
+OTHER_FILES += qmlplugins.qml
 
 
 # Linking

--- a/src/app/app.pro
+++ b/src/app/app.pro
@@ -4,7 +4,7 @@ CONFIG += c++11 warn_on exceptions_off rtti_off
 SOURCES += main.cpp
 DEFINES *= $${COMMON_DEFINES}
 
-RESOURCES += $${TOP_SRCDIR}/assets/assets.qrc
+RESOURCES += "$${TOP_SRCDIR}/assets/assets.qrc"
 OTHER_FILES += qmlplugins.qml
 
 
@@ -17,7 +17,7 @@ include($${TOP_SRCDIR}/src/link_to_frontend.pri)
 # Translations
 
 LOCALE_TS_FILES = $$files($${TOP_SRCDIR}/lang/pegasus_*.ts)
-LOCALE_QRC_IN = $${TOP_SRCDIR}/lang/translations.qrc.in
+LOCALE_QRC_IN = "$${TOP_SRCDIR}/lang/translations.qrc.in"
 
 qtPrepareTool(LRELEASE, lrelease)
 locales.name = Compile translations

--- a/src/link_to_backend.pri
+++ b/src/link_to_backend.pri
@@ -1,7 +1,6 @@
 # Link the project that includes this file to the Backend
 
 QT *= qml quick multimedia svg sql
-CONFIG += c++11 warn_on
 
 win32: LIBS += -luser32 -ladvapi32
 macx: LIBS += -framework Cocoa

--- a/src/link_to_backend.pri
+++ b/src/link_to_backend.pri
@@ -8,18 +8,18 @@ macx: LIBS += -framework Cocoa
 
 # based on the auto-generated code by Qt Creator
 
-win32:CONFIG(release, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/backend/release/ -lbackend
-else:win32:CONFIG(debug, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/backend/debug/ -lbackend
-else:unix: LIBS += -L$${TOP_BUILDDIR}/src/backend/ -lbackend
+win32:CONFIG(release, debug|release): LIBS += "-L$${TOP_BUILDDIR}/src/backend/release/" -lbackend
+else:win32:CONFIG(debug, debug|release): LIBS += "-L$${TOP_BUILDDIR}/src/backend/debug/" -lbackend
+else:unix: LIBS += "-L$${TOP_BUILDDIR}/src/backend/" -lbackend
 
-INCLUDEPATH += $${TOP_SRCDIR}/src/backend $${TOP_SRCDIR}/thirdparty
-DEPENDPATH += $${TOP_SRCDIR}/src/backend $${TOP_SRCDIR}/thirdparty
+INCLUDEPATH += "$${TOP_SRCDIR}/src/backend" "$${TOP_SRCDIR}/thirdparty"
+DEPENDPATH += "$${TOP_SRCDIR}/src/backend" "$${TOP_SRCDIR}/thirdparty"
 
-win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/backend/release/libbackend.a
-else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/backend/debug/libbackend.a
-else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/backend/release/backend.lib
-else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/backend/debug/backend.lib
-else:unix: PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/backend/libbackend.a
+win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/backend/release/libbackend.a"
+else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/backend/debug/libbackend.a"
+else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/backend/release/backend.lib"
+else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/backend/debug/backend.lib"
+else:unix: PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/backend/libbackend.a"
 
 
 # Gamepad backend

--- a/src/link_to_frontend.pri
+++ b/src/link_to_frontend.pri
@@ -1,11 +1,11 @@
 # Link the project that includes this file to the frontend files
 
-win32:CONFIG(release, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/frontend/release/ -lfrontend
-else:win32:CONFIG(debug, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/frontend/debug/ -lfrontend
-else:unix: LIBS += -L$${TOP_BUILDDIR}/src/frontend/ -lfrontend
+win32:CONFIG(release, debug|release): LIBS += "-L$${TOP_BUILDDIR}/src/frontend/release/" -lfrontend
+else:win32:CONFIG(debug, debug|release): LIBS += "-L$${TOP_BUILDDIR}/src/frontend/debug/" -lfrontend
+else:unix: LIBS += "-L$${TOP_BUILDDIR}/src/frontend/" -lfrontend
 
-win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/frontend/release/libfrontend.a
-else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/frontend/debug/libfrontend.a
-else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/frontend/release/frontend.lib
-else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/frontend/debug/frontend.lib
-else:unix: PRE_TARGETDEPS += $${TOP_BUILDDIR}/src/frontend/libfrontend.a
+win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/frontend/release/libfrontend.a"
+else:win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/frontend/debug/libfrontend.a"
+else:win32:!win32-g++:CONFIG(release, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/frontend/release/frontend.lib"
+else:win32:!win32-g++:CONFIG(debug, debug|release): PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/frontend/debug/frontend.lib"
+else:unix: PRE_TARGETDEPS += "$${TOP_BUILDDIR}/src/frontend/libfrontend.a"

--- a/src/link_to_frontend.pri
+++ b/src/link_to_frontend.pri
@@ -1,7 +1,5 @@
 # Link the project that includes this file to the frontend files
 
-CONFIG *= c++11 warn_on
-
 win32:CONFIG(release, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/frontend/release/ -lfrontend
 else:win32:CONFIG(debug, debug|release): LIBS += -L$${TOP_BUILDDIR}/src/frontend/debug/ -lfrontend
 else:unix: LIBS += -L$${TOP_BUILDDIR}/src/frontend/ -lfrontend

--- a/tests/backend/api/api.pro
+++ b/tests/backend/api/api.pro
@@ -1,14 +1,5 @@
-CONFIG += testcase no_testcase_installs
 TARGET = test_Api
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 SOURCES = test_Api.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += $${TOP_SRCDIR}/src/themes/themes.qrc
 
-RESOURCES += \
-    $${TOP_SRCDIR}/src/themes/themes.qrc
-
-
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/api/api.pro
+++ b/tests/backend/api/api.pro
@@ -1,5 +1,5 @@
 TARGET = test_Api
 SOURCES = test_Api.cpp
-RESOURCES += $${TOP_SRCDIR}/src/themes/themes.qrc
+RESOURCES += "$${TOP_SRCDIR}/src/themes/themes.qrc"
 
 include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/configfile/configfile.pro
+++ b/tests/backend/configfile/configfile.pro
@@ -1,13 +1,5 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_ConfigFile
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += configfile.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
-
-RESOURCES += \
-    configfile.qrc
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/collection/collection.pro
+++ b/tests/backend/model/collection/collection.pro
@@ -1,10 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_Collection
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/game/game.pro
+++ b/tests/backend/model/game/game.pro
@@ -1,10 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_Game
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/gameassets/gameassets.pro
+++ b/tests/backend/model/gameassets/gameassets.pro
@@ -1,10 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_GameAssets
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/keyeditor/keyeditor.pro
+++ b/tests/backend/model/keyeditor/keyeditor.pro
@@ -1,12 +1,5 @@
-TEMPLATE = app
-QT += qml
-CONFIG += c++11 warn_on qmltestcase no_testcase_installs
-
 TARGET = test_KeyEditor
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
-
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
 
 OTHER_FILES += \
     tst_add.qml \
@@ -14,3 +7,5 @@ OTHER_FILES += \
     tst_keyname.qml \
     tst_replace.qml \
     utils.js
+
+include($${TOP_SRCDIR}/tests/qmltest_common.pri)

--- a/tests/backend/model/locales/locales.pro
+++ b/tests/backend/model/locales/locales.pro
@@ -1,12 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_Locales
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-#RESOURCES = $${TOP_BUILDDIR}/src/app/lang/translations.qrc
-
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/memory/memory.pro
+++ b/tests/backend/model/memory/memory.pro
@@ -1,9 +1,5 @@
-QT += qml testlib
-CONFIG += c++11 warn_on testcase no_testcase_installs
-
 TARGET = test_Memory
 SOURCES = test_Memory.cpp Container.cpp
 HEADERS = Container.h
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/system/system.pro
+++ b/tests/backend/model/system/system.pro
@@ -1,10 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_System
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/model/themes/themes.pro
+++ b/tests/backend/model/themes/themes.pro
@@ -1,12 +1,6 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_Themes
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
 RESOURCES = $${TOP_SRCDIR}/src/themes/themes.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/processlauncher/processlauncher.pro
+++ b/tests/backend/processlauncher/processlauncher.pro
@@ -1,10 +1,4 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_ProcessLauncher
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/providers/favorites/favorites.pro
+++ b/tests/backend/providers/favorites/favorites.pro
@@ -1,13 +1,5 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_FavoriteDB
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += data/data.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
-
-RESOURCES += \
-    data/data.qrc
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/providers/pegasus/pegasus.pro
+++ b/tests/backend/providers/pegasus/pegasus.pro
@@ -1,13 +1,5 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_PegasusProvider
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += data/data.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
-
-RESOURCES += \
-    data/data.qrc
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/providers/playtime/playtime.pro
+++ b/tests/backend/providers/playtime/playtime.pro
@@ -1,11 +1,5 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = test_Playtime
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 RESOURCES += data.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/backend/utils/utils.pro
+++ b/tests/backend/utils/utils.pro
@@ -1,11 +1,4 @@
-CONFIG += testcase no_testcase_installs
 TARGET = test_Utils
+SOURCES = $${TARGET}.cpp
 
-QT += testlib
-CONFIG += c++11 warn_on exceptions_off
-
-SOURCES = test_Utils.cpp
-DEFINES *= $${COMMON_DEFINES}
-
-
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/benchmarks/configfile/configfile.pro
+++ b/tests/benchmarks/configfile/configfile.pro
@@ -1,13 +1,6 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = bench_ConfigFile
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += data.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)
 
-RESOURCES += \
-    data.qrc

--- a/tests/benchmarks/pegasus_provider/pegasus_provider.pro
+++ b/tests/benchmarks/pegasus_provider/pegasus_provider.pro
@@ -1,13 +1,6 @@
-CONFIG += testcase no_testcase_installs
-
-QT += qml testlib
-CONFIG += c++11 warn_on exceptions_off
-
 TARGET = bench_PegasusProvider
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
+RESOURCES += data/data.qrc
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)
 
-RESOURCES += \
-    data/data.qrc

--- a/tests/cxxtest_common.pri
+++ b/tests/cxxtest_common.pri
@@ -1,0 +1,8 @@
+CONFIG += testcase no_testcase_installs
+CONFIG += c++11 warn_on exceptions_off rtti_off
+
+QT += qml testlib
+
+DEFINES *= $${COMMON_DEFINES}
+
+include($${TOP_SRCDIR}/src/link_to_backend.pri)

--- a/tests/integration/sdl/sdl.pro
+++ b/tests/integration/sdl/sdl.pro
@@ -1,8 +1,4 @@
-CONFIG += testcase no_testcase_installs warn_on
-QT += testlib
-
 TARGET = test_SDL
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
+include($${TOP_SRCDIR}/tests/cxxtest_common.pri)

--- a/tests/integration/sortfilter/sortfilter.pro
+++ b/tests/integration/sortfilter/sortfilter.pro
@@ -1,12 +1,7 @@
-CONFIG += testcase no_testcase_installs
-
-CONFIG += qmltestcase
-
 TARGET = test_SortFilter
 SOURCES = $${TARGET}.cpp
-DEFINES *= $${COMMON_DEFINES}
 
-include($${TOP_SRCDIR}/src/link_to_backend.pri)
-
-DISTFILES += \
+OTHER_FILES += \
     tst_all.qml
+
+include($${TOP_SRCDIR}/tests/qmltest_common.pri)

--- a/tests/qmltest_common.pri
+++ b/tests/qmltest_common.pri
@@ -1,0 +1,8 @@
+CONFIG += qmltestcase no_testcase_installs
+CONFIG += c++11 warn_on exceptions_off rtti_off
+
+QT += qml
+
+DEFINES *= $${COMMON_DEFINES}
+
+include($${TOP_SRCDIR}/src/link_to_backend.pri)

